### PR TITLE
consensus/pbft/backends/simple: add dber to save data into db

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: consensus/pbft/backends/simple/dber_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+type dataStruct struct {
+	Name string `json:"name"`
+}
+
+func TestSaveAndRestore(t *testing.T) {
+	db := newDBer(newTestDb())
+
+	// Test a json struct
+	key := "key"
+	jsonStruct := dataStruct{
```
Comment (alanchchen) [inline consensus/pbft/backends/simple/dber_test.go]: Could you test with a more complex struct?
e.g.,
```
type Foo struct {
    Name string
    Number *big.Int
    Hash common.Hash
}
```

File: consensus/pbft/backends/simple/dber_test.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	if err != nil {
+		t.Error("Should save json struct correctly")
+	}
+
+	result := &dataStruct{}
+	err = db.Restore(key, result)
+	if err != nil {
+		t.Error("Should restore json struct correctly")
+	}
+	if result.Name != jsonStruct.Name {
```
Comment (alanchchen) [inline consensus/pbft/backends/simple/dber_test.go]: Use `reflect.DeepEqual` to compare the structs

File: consensus/pbft/backends/simple/dber_test.go (hunk 1 above)
Comment (markya0616) [inline consensus/pbft/backends/simple/dber_test.go]: OK

File: consensus/pbft/backends/simple/dber_test.go (hunk 2 above)
Comment (markya0616) [inline consensus/pbft/backends/simple/dber_test.go]: OK

File: consensus/pbft/backends/simple/eth_db_test.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		t.Error("Should save the object correctly")
+	}
+
+	result := &Foo{}
+	err = db.Restore(key, result)
+	if err != nil {
+		t.Error("Should restore the object correctly")
+	}
+	if reflect.DeepEqual(foo, result); result.Name != foo.Name {
```
Comment (alanchchen) [inline consensus/pbft/backends/simple/eth_db_test.go]: I think
```
if !reflect.DeepEqual(foo, result) {
    t.Error(...)
}
```
is enough

Comment (markya0616) [inline consensus/pbft/backends/simple/eth_db_test.go]: XDD OK

File: consensus/pbft/backends/simple/eth_db_test.go [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	err = db.Restore(key, &result)
+	if err != nil {
+		t.Error("Should restore the object correctly")
+	}
+	if !reflect.DeepEqual(foo, result) {
+		t.Errorf("Should have the same data, result = %v, expected = %v", result, foo)
+	}
+}
+
+func newTestDb() ethdb.Database {
```
Comment (alanchchen) [inline consensus/pbft/backends/simple/eth_db_test.go]: You could try `ethdb.NewMemDatabase`

**Pull request merged**